app/views/play/CampaignView.coffee

```diff
@@ -622,7 +623,14 @@ module.exports = class CampaignView extends RootView
       level.locked = true if level.requiresSubscription and @requiresSubscription and me.get('hourOfCode')
       level.locked = false if @levelStatusMap[level.slug] in ['started', 'complete']
       level.locked = false if @editorMode
-      level.locked = false if not @course? and @campaign?.get('name') in ['Auditions', 'Intro']
+      if @jqxhr
+        new Promise(@jqxhr.then).then(=>
+          # In certain race condition, orderedLevels are initialized but course is not yet initialized (which is done asynchronously).
+          # In that scenario, the below condition is wrongly met and all levels are unlocked. 
+          # Added this inside a promise so that this executes only after the asynchronous call that initiailizes course is completed  
+          if not @course? and @campaign?.get('name') in ['Auditions', 'Intro']
+            level.locked = false 
+        )
```

Comment:
It seems like we could update this check to be `not @courseInstance?` instead of waiting until the course instance is fetched to proceed here.  This creates quite a few identical promises, that would be nice to avoid.  And, this `level.locked = false` will still never execute because `@course` will be initialized in the other constructor promise before these execute.

It also seems like the first line of this function could be updated to `return if @courseInstance?`, which also seems to intend a hasty exit if we have or are expecting to have a course instance.

Also, it seems like `determineNextLevel` should likewise return if `@courseInstance?`, since we expect to fetch all the course-related objects and use `@applyCourseLogicToLevels`.  Depending on how `determineNextLevel` is used, it seems like we'd falsely apply non-course logic while the course stuff was loading.

Reply:
`annotateLevels` now has async behavior, but I think the callers expect it to only return when all of the levels are processed.  This is no longer true with these changes.